packages/apps/spaces/middleware.ts

```diff
@@ -1,145 +1,184 @@
+                case 422:
+                    console.log('422');
+                    return NextResponse.redirect(
+                        new URL('/api/.ory/ui/login', request.url),
+                    );
+                //return router.push("/login", { query: { aal: "aal2" } })
+                case 401:
+                    console.log('401');
+                    // The user is not logged in, so we redirect them to the login page.
+                    return NextResponse.redirect(
+                        new URL('/api/.ory/ui/login', request.url),
+                    );
+                case 404:
+                    console.log('404');
+                    // the SDK is not configured correctly
+                    // we set this up so you can debug the issue in the browser
+                    return NextResponse.redirect(
+                        new URL('/api/.ory/ui/login', request.url),
+                    );
+                default:
+                    return NextResponse.redirect(
+                        new URL('/api/.ory/ui/login', request.url),
+                    );
+            }
+        });
 }
 
 function getRedirectUrl(
-  userName: string,
-  identityId: string,
-  request: NextRequest,
-) {
-  var newURL = '';
-
-  const requestHeaders = new Headers(request.headers);
-
-  requestHeaders.set('X-Openline-USERNAME', userName);
-  requestHeaders.set('X-Openline-IDENTITY-ID', identityId);
-
-  if (request.nextUrl.pathname.startsWith('/customer-os-api/')) {
-    newURL =
-      process.env.CUSTOMER_OS_API_PATH +
-      '/' +
-      request.nextUrl.pathname.substring('/customer-os-api/'.length);
-
-    requestHeaders.set(
-      'X-Openline-API-KEY',
-      process.env.CUSTOMER_OS_API_KEY as string,
-    );
-  } else if (request.nextUrl.pathname.startsWith('/fs/')) {
-    newURL =
-      process.env.FILE_STORAGE_API_PATH +
-      '/' +
-      request.nextUrl.pathname.substring('/fs/'.length);
-    requestHeaders.set(
-      'X-Openline-API-KEY',
-      process.env.FILE_STORAGE_API_KEY as string,
-    );
-  } else if (request.nextUrl.pathname.startsWith('/sa/')) {
-    newURL =
-      process.env.SETTINGS_API_PATH +
-      '/' +
-      request.nextUrl.pathname.substring('/sa/'.length);
-    requestHeaders.set(
-      'X-Openline-API-KEY',
-      process.env.SETTINGS_API_KEY as string,
-    );
-  } else if (request.nextUrl.pathname.startsWith('/oasis-api/')) {
-    newURL =
-      process.env.OASIS_API_PATH +
-      '/' +
-      request.nextUrl.pathname.substring('/oasis-api/'.length);
-    requestHeaders.set(
-      'X-Openline-API-KEY',
-      process.env.OASIS_API_KEY as string,
-    );
-  }
-
-  if (request.nextUrl.searchParams) {
-    newURL = newURL + '?' + request.nextUrl.searchParams.toString();
-  }
-
-  return NextResponse.rewrite(new URL(newURL, request.url), {
-    request: {
-      headers: requestHeaders,
-    },
-  });
+    userName: string,
```

Comment:
is the styles change wanted? I mean the added spaces

Reply:
auto format using prettier.